xbmc/utils/URIUtils.cpp

```diff
@@ -44,6 +45,69 @@ using namespace XFILE;
 
 const CAdvancedSettings* URIUtils::m_advancedSettings = nullptr;
 
+namespace
+{
+
+constexpr std::string_view DecodeURLSpecialChars{"+%"};
+
+std::optional<char> DecodeOctlet(std::string_view& encoded)
+{
+  if (encoded.length() < 2)
+    return {};
+
+  uint8_t decimal{};
+  const auto res = std::from_chars(&encoded[0], &encoded[2], decimal, 16);
+
+  if (res.ec != std::errc() || res.ptr != &encoded[2])
```

Comment:
For the same reason as above:
```suggestion
  if (res.ec != std::errc() || res.ptr != encoded.data() + 2)
```